include/boost/astronomy/coordinate/base_equatorial_frame.hpp

```diff
@@ -8,239 +8,312 @@
-                }
-
-                //!sets the proper motion in Declination
-                void set_pm_dec(double pm_dec)
-                {
-                    this->motion.set_dlat(pm_dec);
-                }
-
-                //!sets the proper motion in Right Ascension including cos(dec)
-                void set_pm_ra_cosdec(double pm_ra_cosdec)
-                {
-                    this->motion.set_dlon_coslat(pm_ra_cosdec);
-                }
-
-                //!sets the radial_velocity
-                void set_radial_velocity(double radial_velocity)
-                {
-                    this->motion.set_ddist(radial_velocity);
-                }
-
-                //!set value of motion including cos(dec)
-                void set_pm_dec_ra_radial(double pm_dec, double pm_ra_cosdec, double radial_velocity)
-                {
-                    this->motion.set_dlat_dlon_coslat_ddist(pm_dec, pm_ra_cosdec, radial_velocity);
-                }
-            };
-
-        } //namespace coordinate
-    } //namespace astronomy
-} //namespace boost
+    ///@cond INTERNAL
+    BOOST_STATIC_ASSERT_MSG((std::is_base_of
+            <spherical_representation<typename Representation::type, 
+            typename Representation::quantity1, typename Representation::quantity2,
+            typename Representation::quantity3>, Representation>::value),
+            "argument type is expected to be a spherical_representation class");
+    BOOST_STATIC_ASSERT_MSG((std::is_base_of
+            <spherical_coslat_differential<typename Differential::type, 
+            typename Differential::quantity1, typename Differential::quantity2,
+            typename Differential::quantity3>, Differential>::value),
+            "argument type is expected to be a spherical_coslat_differential class");
+    ///@endcond
+
+public:
+    typedef typename Representation::type RepCoordinateType;
+    typedef typename Representation::quantity1 RepLatQuantity;
+    typedef typename Representation::quantity2 RepLonQuantity;
+    typedef typename Representation::quantity3 RepDistQuantity;
+    typedef typename Differential::type DifCoordinateType;
+    typedef typename Differential::quantity1 DifLatQuantity;
+    typedef typename Differential::quantity2 DifLonQuantity;
+    typedef typename Differential::quantity3 DifDistQuantity;
+    
+    //default constructor no initialization
+    base_equatorial_frame() {}
+
+    //!constructs object from another representation object
+    template <typename OtherRepresentation>
+    base_equatorial_frame(OtherRepresentation const& representation_data)
+    {
+        BOOST_STATIC_ASSERT_MSG((boost::astronomy::detail::is_base_template_of
+            <boost::astronomy::coordinate::base_representation, OtherRepresentation>::value),
+            "argument type is expected to be a representation class");
+        
+        auto temp = make_spherical_representation<RepCoordinateType, RepLatQuantity,
```

Comment:
I think we need to use all 8 arguments as if we use 4 arguments and when `Return` types are same as the types of the `base_equatorial_frame` class then there would be error `call of overloaded` function because then two `make` functions become same; the one having 8 arguments and other the `make` function which just creates a copy.

Reply:
In such case, we need to reconsider the implementation of that `make` function 8 arguments are too much for the user and 4 of them are actually repeating.

Reply:
But user will actually know the coordinates and their quantities beforehead. So he/she would have to use:
- either 0 parameters when he/she would just be copying the point 
- or 4 parameters when he/she would have to return some other quantity

Reply:
yes, that's what I had in my mind when I created the function then why are you using 8? does it give any error if we use 4 parameters there?

Reply:
Yes there is no problem in 4 parameters when user uses it as he already know the quantities. But here we don't know whether `OtherRepresentation` is same as `Representation` or it is different. If we use 4 parameters and it is same then it will give an error `call of overloaded function`

Reply:
Ok understood, my mistake. For internal use, we'll prefer to use 8 template parameters.